myapp/app/models/task.rb

```diff
@@ -2,6 +2,17 @@ class Task < ApplicationRecord
   belongs_to :group
 
   validates :name, presence: true
+  validates :priority, presence: true
+  validates :group_id, presence: true
+  validate :verify_end_period_at
 
   enum priority: { high: 0, middle: 1, low: 2 }
+
+  private
+
+  def verify_end_period_at
+    return if end_period_at.blank? || Date.current <= end_period_at.to_date
```

Comment:
あ、すみません、以下でいいと思うんですけどどうですか？
```suggestion
    return if end_period_at.blank? || Date.current <= end_period_at
```

Reply:
すみません。修正しました。end_period_atのデータ型がdatetimeだったので、それもdateに変えておきました。

Reply:
むむ、あれ、データ型まで変える必要あったんですかね？
変えても全然いいとは思うんですけど、上記コードを反映するだけでも意図した挙動になったような気がしますー
私は単純に、`to_date` メソッドがなくても全く同じ挙動をするので、不要なコードは消してしまったほうがいいかなと思って指摘した感じでした！

Reply:
@kenkaton to_dateを無くすだけでも、全然問題ないです！！！

ただし、
DateとDatetime、データ型が違うのに比較すると将来にミス起きやすいところになるかとと思いましたので、統一しました！！

Reply:
なるほどー、了解しました！
ちょっと懸念点としては、`end_period_at` というカラム名（datetime型かなと思ってしまう）と実際に入っているデータ（date型が入っている）が少し一致していないところが気になりましたー
ですが、動作に影響はないので問題ないと思います:+1:

Reply:
そうなんですね！！！英語が弱くて分からなかったです！！